### PR TITLE
Normalize license across the repo

### LINKED ISSUE #4807: Normalize rust `Cargo.toml` license field
`Cargo.toml` with a license field:

| File                                                | License     |
| --------------------------------------------------- | ----------- |
| `libparsec/crates/client/Cargo.toml`                | ` BUSL-1.1` |
| `libparsec/crates/client_high_level_api/Cargo.toml` | ` BUSL-1.1` |
| `libparsec/crates/crypto/Cargo.toml`                | ` BUSL-1.1` |
| `libparsec/crates/protocol/Cargo.toml`              | ` BUSL-1.1` |
| `libparsec/crates/serialization_format/Cargo.toml`  | ` BUSL-1.1` |
| `libparsec/crates/tests_macros/Cargo.toml`          | ` BUSL-1.1` |
| `libparsec/crates/types/Cargo.toml`                 | ` BUSL-1.1` |

> The possible whitespace in the license value are from the `Cargo.toml` value.

`Cargo.toml` without a license field:

- `Cargo.toml`
- `bindings/android/libparsec/rust/Cargo.toml`
- `bindings/electron/Cargo.toml`
- `bindings/web/Cargo.toml`
- `libparsec/Cargo.toml`
- `libparsec/crates/client_connection/Cargo.toml`
- `libparsec/crates/platform_async/Cargo.toml`
- `libparsec/crates/platform_device_loader/Cargo.toml`
- `libparsec/crates/platform_http_proxy/Cargo.toml`
- `libparsec/crates/platform_storage/Cargo.toml`
- `libparsec/crates/testbed/Cargo.toml`
- `libparsec/crates/tests_fixtures/Cargo.toml`
- `server/Cargo.toml`

The goal woulb be to normalize that value, so the file without a license field should have one.

Since our current header is on each source file:

```
// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
```

We should use the value `BUSL-1.1 OR AGPL-3.0` as for the value of the `license` field

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
Enfore specific license value for:

- `rust` (`Cargo.toml`)
- `python` (`pyproject` & poetry mode)
- `npm` (`package.json`)

Closes #4807

### CODE REVIEW

File: misc/version_updater.py [hunk 1]
```diff
@@ -70,6 +74,7 @@ class Tool(enum.Enum):
     Tool.Node: "18.12.0",
     Tool.WasmPack: "0.11.0",
     Tool.Parsec: "2.16.0-a.0+dev",
+    Tool.License: "BUSL-1.1 OR AGPL-3.0-only",
```
Comment (FirelightFlagboy) [inline misc/version_updater.py:77]: @touilleMan I've choose to use the value `BUSL-1.1 OR AGPL-3.0-only`, should I keep it that way or you prefer to directly move to `BUSL-1.1` only ?

Comment (touilleMan) [reply misc/version_updater.py:77]: Yeah move on BUSL-1.1 directly

Comment (FirelightFlagboy) [reply misc/version_updater.py:77]: So I need to also update the headers ?

File: misc/bench.py [hunk 2]
```diff
@@ -1,6 +1,4 @@
 #!/usr/bin/env python3
-# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPL-3.0 2016-present Scille SAS
-
```
Comment (FirelightFlagboy) [inline misc/bench.py:3]: Why this file is excluded from `pre-commit` ?

File: misc/license_headers.py [hunk 3]
```diff
@@ -1,6 +1,4 @@
 #!/usr/bin/env python
-# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPL-3.0 2016-present Scille SAS
-
```
Comment (FirelightFlagboy) [inline misc/license_headers.py:3]: `license_header` configure itself to not have a header, should I revert the deletion ? should I change the header to `BUSL-1.1` ?

Comment (mmmarcos) [reply misc/license_headers.py:3]: I am not sure I understand the question. :thinking: Do you mean that you remove it because the file sets its own header or something like that? 
In any case, I think it is probably better/clearer to set it to `BUSL-1.1`

File: server/parsec/backend/utils.py [hunk 4]
```diff
@@ -26,8 +27,9 @@
     from parsec.backend.client_context import BaseClientContext
 
     Ctx = TypeVar("Ctx", bound=BaseClientContext)
-    T = TypeVar("T")
-    P = ParamSpec("P")
+
+T = TypeVar("T")
+P = ParamSpec("P")
```
Comment (mmmarcos) [inline server/parsec/backend/utils.py:32]: This does not seem related to the license change, was it the formatter? Is the indentation change wanted?

File: server/parsec/monitoring.py [hunk 5]
```diff
@@ -1,5 +1,5 @@
-# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPL-3.0 2016-present Scille SAS
-from __future__ import annotations
+# Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
```
Comment (mmmarcos) [inline server/parsec/monitoring.py:2]: Is this a wanted change?

**Review by mmmarcos — COMMENTED** [review]
All in all it LGTM, just a couple of comments that should be checked.

For example, on many files the line `from __future__ import annotations` is removed when is next to the license line (for example [server/parsec/serde/serializer.py](https://github.com/Scille/parsec-cloud/pull/4822/files#diff-3f01dba235d54091f986e72d3454b1c786fe178d8a5334d6535c79e4caf681a5)). Is that a wanted change? (you can grep on the commit diff to identify such lines)

**Comment by FirelightFlagboy** [comment]
Superseded by #4947
